core/src/database/base-repository.ts

```diff
@@ -0,0 +1,21 @@
+import {Injectable} from "@nestjs/common";
+import type {DeepPartial} from "typeorm";
+import {
+    DataSource, EntityTarget, Repository,
+} from "typeorm";
+
+import type {BaseModel} from "./base-model";
+
+@Injectable()
+export abstract class BaseRepository<T extends BaseModel> extends Repository<T> {
+    constructor(C: EntityTarget<T>, dataSource: DataSource) {
+        super(C, dataSource.createEntityManager());
+    }
+
+    async createAndSave(data: DeepPartial<T>): Promise<T> {
```

Comment:
DeepPartial is what TypeORM uses in their create functions. Without it being a partial, it would require every single field (id, timestamps, every single possible relationship). I just copied their function header.

Reply:
That's what I mean; we can use Omit, but then for each repository, we need to do that manually and create the type for each one and then update it if we ever update the schemas.